### PR TITLE
[RDY] Remove -Z flag from --help message

### LINKED ISSUE #13731: nvim --help shows -Z flag, which has been removed
- `nvim --version`: NVIM v0.5.0-dev+1012-gfe1ebea33
- `vim -u DEFAULTS` (version: ) behaves differently? Not applicable, since `vim` retains `-Z`
- Operating system/version: Ubuntu 20.04.1 LTS
- Terminal name/version: Konsole 19.12.3
- `$TERM`: xterm-256color

`nvim --help` shows the `-Z` flag, which was removed in PR #11996.

In PR #13338, a lingering reference was removed from the man page.

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC --help | grep '\-Z'
```

### Actual behaviour

Output:
```
  -Z                    Restricted mode
```

### Expected behaviour

No output

### PR DESCRIPTION
**Pull request by srafi1**
Restricted mode (-Z) was removed in #11996 and lingering references to it in the man pages were removed in #13338. This PR removes its reference from the `--help` message and removes all translations of that line from `src/nvim/po/*.po` files. Fixes #13731.

### CODE REVIEW

**Review by janlazo — APPROVED** [review]

**Comment by janlazo** [comment]
Will merge if this PR is ready.

**Comment by srafi1** [comment]
@janlazo I think it's ready. This is my first PR on neovim, so I was waiting on confirmation from someone else that I didn't miss anything, but I think it's ready after having searched for `-Z` and `Restricted mode` using ripgrep.

**Comment by MdAbedin** [comment]
@srafi1 nice pull request bro

**Pull request merged**
